provider/coredns/coredns.go

```diff
@@ -148,10 +177,53 @@ func (c etcdClient) DeleteService(ctx context.Context, key string) error {
 	ctx, cancel := context.WithTimeout(ctx, etcdTimeout)
 	defer cancel()
 
+	if owned, err := c.isOwnedBy(ctx, key); err != nil {
+		return err
+	} else if !owned {
+		return fmt.Errorf("key %q is not owned by this service", key)
+	}
+
 	_, err := c.client.Delete(ctx, key, etcdcv3.WithPrefix())
 	return err
 }
 
+func (c etcdClient) isOwnedBy(ctx context.Context, key string) (bool, error) {
+	if !c.strictlyOwned {
+		return true, nil
+	}
+
+	r, err := c.client.Get(ctx, key)
```

Comment:
```suggestion
	r, err := c.client.Get(ctx, key, etcdcv3.WithLimit(1))
```

Reply:
Is this required? If you are not using prefixkeys, it should always return a single entry.